### PR TITLE
new: grind-timer

### PR DESCRIPTION
**Pull request by kiwec**
livesplit-inspired timer for pp farmers (or content creators)

i'm too lazy to make a gif but it's exactly what you expect, it turns green and goes up

grind-timer by kiwec

### CODE REVIEW

File: counters/grind-timer by kiwec/index.html [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        const current_time = Date.now();
+
+        if(tosu.beatmap.checksum != bmap_md5) {
+          bmap_md5 = tosu.beatmap.checksum;
+          start_time = current_time;
+          pause_time = current_time;
+          timer.classList.add('paused');
+        }
+
+        if(pause_time == 0) {
```
Comment (cyperdark) [inline counters/grind-timer by kiwec/index.html:87]: Feature request: add settings option to all count time outside play

https://github.com/tosuapp/tosu/wiki#settings-api

File: counters/grind-timer by kiwec/index.html [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        text += milliseconds.toString().padStart(2, '0');
+
+        timer.textContent = text;
+
+        requestAnimationFrame(tick);
+      }
+
+      tick();
+
+      const socket = new WebSocket("ws://127.0.0.1:24050/websocket/v2");
```
Comment (cyperdark) [inline counters/grind-timer by kiwec/index.html:75]: preferable add reconnecting websocket, or socket.js:
https://github.com/tosuapp/tosu/wiki#socket-wrapper

**Review by cyperdark — CHANGES_REQUESTED** [review]

File: counters/grind-timer by kiwec/index.html (hunk 1 above)
Comment (GabuTheDev) [reply counters/grind-timer by kiwec/index.html]: > Can you clarify what this means? It already pauses the timer whenever you're not actively playing the map (except the pause screen, I didn't see any keys in the JSON for it).

Ck probably wants an option to make the timer always run, no matter the screen status. (Yours currently, as stated in the description the overlay, resets it's time when the user switches to another beatmap. Ck wants it to always count time)

Comment (cyperdark) [reply counters/grind-timer by kiwec/index.html]: ^ yeah, and setting for it will be nice to have

**Comment by cyperdark** [comment]
oh shit i forgot, ping me tomorrow in discord

**Review by cyperdark — APPROVED** [review]

**Pull request merged**